common/core/src/endpoint.ts

```diff
@@ -6,16 +6,20 @@
 
 export const apiVersion = '2017-11-08-preview';
 
-export function devicePath(id: string): string {
-  return '/devices/' + id;
+export function devicePath(deviceId: string, moduleId?: string): string {
```

Comment:
>devicePath [](start = 16, length = 10)

I see why (simpler for the callers). yet it's not aptly named and that does not make me too happy regarding separation of concerns.

I'd rather have 2 files: deviceEndpoints and moduleEndpoints. DeviceEndpoint stays unchanged, moduleEndpoints calls deviceEndpoints.
or something along those lines that clearly separate things.

Also, it's a good time to revisit how we manage endpoints. many parts of the code don't even use these. since we have time to refine code, i'd like to see this refactored.